File: solutions/clojure/resistor-color/1/src/resistor_color.clj
(ns resistor-color)

(def color-map {"black" 0, "brown" 1, "red" 2, "orange" 3, 
                "yellow" 4, "green" 5, "blue" 6, "violet" 7,
                "grey" 8, "white" 9})

(def colors
 (vec (keys (into (sorted-map-by 
             (fn [k1 k2] (compare (get color-map k1)
               (get color-map k2)))) color-map))))

(defn color-code
  "Returns the numerical value associated with the given color"
  [color]
  (get color-map color))

File: solutions/free-pascal/resistor-color/1/ResistorColor.pas
unit ResistorColor;

{$mode ObjFPC}{$H+}

interface

type
  TStrArray = Array Of String;

function ColorCode(const color : string) : int8;
function colors : TStrArray;

implementation

uses SysUtils;

function ColorCode(const color : string) : int8;
var 
  AllColors : TStrArray;
  i : integer;
begin
    result := -1;
    AllColors := colors();
    for i:=0 to Length(AllColors) do
    begin
     if AllColors[i] = color then
       begin
         result := i;
         break;
       end;
    end;
      
end;

function colors : TStrArray;
begin

 result := ['black', 'brown', 'red', 'orange', 'yellow', 'green', 'blue', 'violet', 'grey', 'white']

end;

end.
